scripts/hcp_glm.py

```diff
@@ -0,0 +1,209 @@
+    contrasts = dict((cid, [cval] * len(fmri_files))
+                     for cid, cval in contrasts.iteritems())
+
+    print (('Fitting a "Fixed Effect" GLM for merging LR and RL '
+            'phase-encoding directions for subject %s (%s task)...' % (
+                subject_id, task)))
+    fmri_glm = FirstLevelGLM(memory=memory, smoothing_fwhm=smoothing_fwhm)
+    fmri_glm.fit(fmri_files, design_matrices)
+
+    # save computed mask
+    mask_path = os.path.join(subject_output_dir, "mask.nii")
+    print("Saving mask image to %s ..." % mask_path)
+    fmri_glm.masker_.mask_img_.to_filename(mask_path)
+
+    # do the actual model fit
+    z_maps = {}
+    effects_maps = {}
+    map_dirs = {}
+    for contrast_id, contrast_val in contrasts.iteritems():
+        print("\tcontrast id: %s" % contrast_id)
+        z_map, eff_map = fmri_glm.transform(
+            contrast_val, contrast_name=contrast_id, output_z=True,
+            output_effects=True)
+
+        # store stat maps to disk
+        for map_type, out_map in zip(['z', 'effects'],
+                                     [z_map, eff_map]):
+            map_dir = os.path.join(
+                subject_output_dir, '%s_maps' % map_type)
+            map_dirs[map_type] = map_dir
+            if not os.path.exists(map_dir):
+                os.makedirs(map_dir)
+            map_path = os.path.join(map_dir, '%s_%s.nii' % (map_type,
+                                                            contrast_id))
+            print("\t\tWriting %s ..." % map_path)
+            nibabel.save(out_map, map_path)
+
+            # collect zmaps for contrasts we're interested in
+            if map_type == 'z':
+                z_maps[contrast_id] = map_path
+
+            if map_type == 'effects':
+                effects_maps[contrast_id] = map_path
+
+    if report:
+        stats_report_filename = os.path.join(subject_output_dir, "reports",
+                                             "report_stats.html")
+        generate_subject_stats_report(
+            stats_report_filename, contrasts, z_maps,
+            fmri_glm.masker_.mask_img_, threshold=2.3, cluster_th=15,
+            design_matrices=design_matrices, TR=tr, subject_id=subject_id,
+            start_time=stats_start_time, n_scans=n_scans, paradigm=paradigm,
+            frametimes=frametimes, drift_model=drift_model, hfcut=hfcut,
+            title="GLM for subject %s" % subject_id, hrf_model=hrf_model,
+            anat=anat_file)
+        ProgressReport().finish_dir(subject_output_dir)
+        print("Statistic report written to %s\r\n" % stats_report_filename)
+
+    print ("Done (subject %s)" % subject_id)
+    return dict(subject_id=subject_id, mask=mask_path,
+                effects_maps=effects_maps, z_maps=z_maps,
+                contrasts=contrasts)
+
+
+if __name__ == "__main__":
+    # get subjects to process
+    subject_dirs = sorted(glob.glob(os.path.join(
+        root, "storage/data/HCP/S500-1/??????")))[:n_subjects]
```

Comment:
Note also that there are some right issueson drago
IOError: [Errno 13] Permission denied: '/storage/workspace/elvis/HCP_GLM/cache_dir/100307/joblib/nilearn/masking/compute_epi_mask/func_code.py'